### DISCUSSION TITLE
Handling actions of nested store in testing

### DISCUSSION THREAD

**Discussion by 2sem** [pointfreeco/swift-composable-architecture#3111] [Q&A]
Hey iOS Devs! how to solve this key path problem? Help!

I want to know whether I can't handle specific nested action in testing.
```await store.receive(\.addDateAction.delegate)``` works
But I want to handle only saveDate

Thanks to your help.

<img width="828" alt="스크린샷 2024-05-23 오후 4 22 17" src="https://github.com/pointfreeco/swift-composable-architecture/assets/16129260/b820fb8f-6575-42a3-b20d-516245cb3fbf">

**Comment by mbrandonw** [comment]
Note that you are using key path syntax `\.addDateAction.delegate.saveData`, and so it is not valid syntax to put parentheses at the end as you would with enum cases. Instead you provide the payload as a separate argument to `store.receive`:

```
await store.receive(\.addDateAction.delegate.saveData, Date(…))
```

**Comment by 2sem** [comment]
@CasePathable
TCA 1.9.3
